### PR TITLE
fix: eslint and typescript issues

### PR DESCRIPTION
**Pull request by deuchainn**
What's done here 

## Eslint Fixes

Ran `pnpm lint:eslint --fix` first from the `packages/client` directory 👇 :
Screenshot 2024-09-10 at 22 49 10

Which autofixed what it could.

## TypeScript fixes

Ran `pnpm lint:typecheck` from the `packages/client` directory 👇 :
Screenshot 2024-09-10 at 22 50 40

Found 22 errors ☝️ , and fixed them manually.

### CODE REVIEW

File: packages/client/src/Frontend/Components/BrowserIssues.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import type { Incompatibility } from "@frontend/Utils/BrowserChecks";
+import { Incompatibility } from "@frontend/Utils/BrowserChecks";
```
Comment (deuchainn) [inline packages/client/src/Frontend/Components/BrowserIssues.tsx:1]: In this file the `Incompatibility` is used as a value and not a type.

File: packages/client/src/Frontend/Pages/GameLandingPage.tsx
Comment (deuchainn) [inline packages/client/src/Frontend/Pages/GameLandingPage.tsx:1]: In this file we had many unused values, we just comment these out.

We could disable this error reporting.  However from experience at work, not having error reporting on this makes it difficult to see what's used or not, and one end not cleaning up unused code. It's better to go back in GIT history and see what was the code used if one needs to add back some code instad.

File: packages/client/src/_types/global/GlobalTypes.ts [hunk 2]
```diff
@@ -1,9 +1,8 @@
+import type { GameManager } from "@backend/GameLogic/GameManager";
+import type { GameUIManager } from "@backend/GameLogic/GameUIManager";
 import type { Rectangle } from "@df/types";
 import type { Dispatch, SetStateAction } from "react";
 
-import type GameManager from "../../Backend/GameLogic/GameManager";
-import type GameUIManager from "../../Backend/GameLogic/GameUIManager";
```
Comment (deuchainn) [inline packages/client/src/_types/global/GlobalTypes.ts:5]: Here we are trying to import defaults in ESM one usually don't export defaults 🤷

File: packages/client/src/_types/global/GlobalTypes.ts [hunk 3]
```diff
@@ -1,9 +1,8 @@
+import type { GameManager } from "@backend/GameLogic/GameManager";
+import type { GameUIManager } from "@backend/GameLogic/GameUIManager";
```
Comment (deuchainn) [inline packages/client/src/_types/global/GlobalTypes.ts:2]: Importing the correct exported classes, and also from the aliased `@backed` package.

**Review by fromddy — APPROVED** [review]

**Pull request merged**